src/_gettsim/taxes/lohnst.py

```diff
@@ -0,0 +1,401 @@
+
+    # b) Take the actual contributions (usually the better option),
+    #   but apply the reduced rate!
+
+    if ges_pflegev_zusatz_kinderlos:
+        beitr_satz_pflegev = (
+            soz_vers_beitr_params["beitr_satz"]["ges_pflegev"]["standard"]
+            + soz_vers_beitr_params["beitr_satz"]["ges_pflegev"]["zusatz_kinderlos"]
+        )
+    else:
+        beitr_satz_pflegev = soz_vers_beitr_params["beitr_satz"]["ges_pflegev"][
+            "standard"
+        ]
+
+    vorsorg_kv_option_b = bruttolohn_kv * (
+        soz_vers_beitr_params["beitr_satz"]["ges_krankenv"]["ermäßigt"] / 2
+        + ges_krankenv_zusatzbeitrag / 2 / 100
+        + beitr_satz_pflegev
+    )
+
+    # add both RV and KV deductions. For KV, take the larger amount.
+    out = vorsorg_rv + max(vorsorg_kv_option_a, vorsorg_kv_option_b)
+    return out
+
+
+@add_rounding_spec(params_key="lohnst")
+def vorsorgepauschale_ab_2005_bis_2009() -> float:
+    """vorsorg_rv and vorsorg_kv_option_a are identical to after 2010."""
+
+    out = 0
+    return out
+
+
+def kinderfreib_für_soli_st_lohnst(
+    steuerklasse: int,
+    anz_kinder_mit_kindergeld_tu: float,
+    eink_st_abzuege_params: dict,
+) -> float:
+    """Calculate Child Allowance for Lohnsteuer-Soli.
+
+    For the purpose of Soli on Lohnsteuer, the child allowance not only depends on the
+    number of children, but also on the steuerklasse
+
+    """
+
+    kinderfreib_basis = (
+        eink_st_abzuege_params["kinderfreib"]["sächl_existenzmin"]
+        + eink_st_abzuege_params["kinderfreib"]["beitr_erz_ausb"]
+    )
+
+    # For certain tax brackets, twice the child allowance can be deducted
+    if steuerklasse == 1 | steuerklasse == 2 | steuerklasse == 3:
+        out = kinderfreib_basis * 2 * anz_kinder_mit_kindergeld_tu
+    elif steuerklasse == 4:
+        out = kinderfreib_basis * anz_kinder_mit_kindergeld_tu
+    else:
+        out = 0
+    return out
+
+
+def lohnst_mit_kinderfreib(
+    lohnst_eink: float,
+    kinderfreib_für_soli_st_lohnst: float,
+    steuerklasse: int,
+    eink_st_params: dict,
+) -> float:
+    """Calculate Lohnsteuer with the tax base taking child allowance into account,
+    otherwise identical to :func:`lohnst_m`.
+
+    """
+
+    eink = max(lohnst_eink - kinderfreib_für_soli_st_lohnst, 0)
+    lohnsteuer_basistarif = _eink_st_tarif(eink, eink_st_params)
+    lohnsteuer_splittingtarif = 2 * _eink_st_tarif(eink / 2, eink_st_params)
+    lohnsteuer_klasse5_6 = max(
+        2
+        * (
+            _eink_st_tarif(eink * 1.25, eink_st_params)
+            - _eink_st_tarif(eink * 0.75, eink_st_params)
+        ),
+        eink * eink_st_params["eink_st_tarif"]["rates"][0][1],
+    )
+
+    if steuerklasse in (1, 2, 4):
+        out = lohnsteuer_basistarif
+    elif steuerklasse == 3:
+        out = lohnsteuer_splittingtarif
+    else:
+        out = lohnsteuer_klasse5_6
+
+    return out
+
+
+# Possible ToDo: Determine Steuerklasse endogenously.
```

Comment:
I am not sure we'll ever want this, seems to depend a lot on assumptions.

In any case, please get rid of the code snippet here and in case you think this should be discussed more, open an issue for it. You can paste the code there then.

Reply:
Please post the code in that issue. This will be very helpful for the data preparation.